src/components/Tooltip.js

```diff
@@ -0,0 +1,135 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Animated, Text, View,
+} from 'react-native';
+import Hoverable from './Hoverable';
+import getTooltipStyles from '../styles/getTooltipStyles';
+
+const propTypes = {
+    // The text to display in the tooltip.
+    text: PropTypes.string.isRequired,
+
+    // Child to wrap with Tooltip. Must be only child.
+    children: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.element,
+        PropTypes.instanceOf(Component),
+    ]).isRequired,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.animation = new Animated.Value(0);
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+        this.width = 0;
+        this.height = 0;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+        this.tooltipWidth = 0;
+        this.tooltipHeight = 0;
+
+        // The distance between the left side of the rendered view and the left side of the window
+        this.xOffset = 0;
+
+        // The distance between the top of the rendered view and the top of the window
+        this.yOffset = 0;
+
+        this.getPosition = this.getPosition.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    /**
+     * Get the position and measure the size of the wrapper view and the tooltip itself.
+     */
+    getPosition() {
```

Comment:
This could maybe be renamed to `measureViews` or `measureWrapperAndTooltipViews` and then the purpose of the method might be more intuitive.